dwave/plugins/torch/boltzmann_machine.py

```diff
@@ -292,16 +299,21 @@ def sample(
                 If ``None`` and data is not a tensor then the result tensor is
                 constructed on the current device.
             sample_params (dict, optional): Parameters of the `sampler.sample` method.
+            as_tensor (bool): Whether to return the sampleset as a tensor.
+                Defaults to ``True``. If ``False`` returns a ``dimod.SampleSet``.
 
         Returns:
-            torch.Tensor: Spins sampled from the model
+            torch.Tensor | SampleSet: Spins sampled from the model
             (shape prescribed by ``sampler`` and ``sample_params``).
         """
         if sample_params is None:
             sample_params = dict()
         h, J = self.to_ising(prefactor, linear_range, quadratic_range)
-        ss = spread(sampler.sample_ising(h, J, **sample_params))
-        spins = self.sampleset_to_tensor(ss, device=device)
+        spins = spread(sampler.sample_ising(h, J, **sample_params))
```

Comment:
The variable name "sample_set" would be more explicit than "spins" in describing its object/type